yoti-connect/yoti-connect.php

```diff
@@ -139,7 +139,7 @@ function yoti_connect_login($user_login=null, $user=null)
     }
 
     $activityDetails = YotiConnectHelper::getYotiUserFromStore();
-    if ($activityDetails && empty($_SESSION['yoti_nolink']))
+    if ($activityDetails && (!isset($_POST['yoti_nolink']) || empty($_POST['yoti_nolink'])))
```

Comment:
I think not empty should be fine, because this is checkbox value we are checking which in some case could be boolean or numeric true or 1. checking if it's not empty should be fine.